File: scratch/update_analytics.js
const fs = require('fs');
const file = 'services/portal-api/internal/transport/http/handler/analytics.go';
let content = fs.readFileSync(file, 'utf8');

const regex = /func \(h \*AnalyticsHandler\) HandleGetStatistics\(w http\.ResponseWriter, r \*http\.Request\) \{[\s\S]*\}\n/m;
const repl = fs.readFileSync('scratch/prom_logic.go', 'utf8');

content = content.replace(regex, 'func (h *AnalyticsHandler) HandleGetStatistics(w http.ResponseWriter, r *http.Request) {\n' + repl + '\n');
fs.writeFileSync(file, content);
console.log('done replacing analytics.go');

File: scratch/update_analytics.py
import re

with open('services/portal-api/internal/transport/http/handler/analytics.go', 'r') as f:
    content = f.read()

with open('scratch/prom_logic.go', 'r') as f:
    prom_logic = f.read()

# Replace from HandleGetStatistics all the way to the end
pattern = re.compile(r'func \(h \*AnalyticsHandler\) HandleGetStatistics\(w http\.ResponseWriter, r \*http\.Request\) \{.*', re.DOTALL)
content = pattern.sub(f'func (h *AnalyticsHandler) HandleGetStatistics(w http.ResponseWriter, r *http.Request) {{\n{prom_logic}', content)

with open('services/portal-api/internal/transport/http/handler/analytics.go', 'w') as f:
    f.write(content)

print("done")
